### DISCUSSION TITLE
Separate healthy from neoplastic cells using 5'-BCR seq.

### DISCUSSION THREAD

**Discussion by cv55** [tuonglab/dandelion#379] [Q&A]
Hello, thanks for the great tool! Our main goal is to separate cancer from healthy cells in our samples. We expect cancer cells to form clusters that consist of one clone and healthy cells to form clusters that encompass multiple clones. In one sample I get this picture:
![Screenshot 2024-04-29 at 09 55 41](https://github.com/zktuong/dandelion/assets/18632504/eb0aa40b-b85f-44fd-888b-ec03ac43296e)
How safe would it be to say the cluster that has many no_contig cells is the healthy one? 
I know that these are barcodes that do not match between the GEX and BCR assays, but it seems to me that the big cluster is a unique clone. 

Do you have any suggestions on how to separate cancer and healthy cells using Dandelion?

**Comment by zktuong** [comment]
Hi @cv55 , thanks for the interest in Dandelion! 

Is your cancer a B-cell neoplasm and are those cells with no contigs B cells? Also it seems to me that some of the colours are overlapping. How does it look if the colours are more distinct?

I think it also depends on how you set up your umap here - did you remove VDJ genes from the highly variable features before you scale/regress/pca/neighbors/umap? if not, maybe remove them first and try again.

You might want to consider orthogonal ways of supporting this observation e.g. [infercnvpy](https://github.com/icbi-lab/infercnvpy) to examine if those no contig cells indeed have little/no CNV events

**Reply by cv55** [reply]
Hello, thanks for the reply. 
I will make my question a bit more general. These cells have been annotated as B-cells using the expression of typical marker genes using the 5' GEX assay. However, some of these cells do not have reads spanning the VDJ or VJ regions. Is this possible? Could these be doublets, or is it likely that these are B-cells for which BCR information was not retrieved because of biological reasons?

**Reply by zktuong** [reply]
Hi, it's probably due to technical reasons (drop outs or low reads) rather than biological if they are mature B cell.

**Reply by cv55** [reply]
Thanks a lot, one more question. In some cases I have cells that are marked as 'single pair' but when i check the VDJ genes I get something like this: IGHV3-23,IGHV3-21. If I interpret this correctly this cell expresses two V genes. Is this not an indication that this cell has an extra pair of heavy chains? Or this could be biological, eg due to recombination?
--


<br class="Apple-interchange-newline">

**Reply by zktuong** [reply]
ah this one doesn't mean it's 2 V genes. It means that there are multiple equivalent top matches, and igblast reports them by separating with a comma. I think most people use the first annotated gene for simplicity (or remove the contig if necessary).
